substrate/frame/asset-conversion/src/lib.rs

```diff
@@ -713,33 +701,43 @@ pub mod pallet {
 		/// respecting `keep_alive`.
 		///
 		/// If successful, returns the amount of `path[1]` acquired for the `amount_in`.
-		pub fn do_swap_exact_tokens_for_tokens(
+		///
+		/// WARNING: This may return an error after a partial storage mutation. It should be used
+		/// only inside a transactional storage context and an Err result must imply a storage
+		/// rollback.
+		pub(crate) fn do_swap_exact_tokens_for_tokens(
 			sender: T::AccountId,
-			path: BoundedVec<T::MultiAssetId, T::MaxSwapPathLength>,
-			amount_in: T::AssetBalance,
-			amount_out_min: Option<T::AssetBalance>,
+			path: Vec<T::MultiAssetId>,
```

Comment:
How would you connect this to xcm later? Don't the safe call filters prevent any unbounded extrinsics?

Reply:
we should be good now, since we starting to use the message queue 
https://github.com/paritytech/polkadot-sdk/pull/1246